### PR TITLE
Aromatic bond display option

### PR DESCRIPTION
**Pull request by arose**
This adds an `aromaticBonds` option to the Ball & Stick representation defaulting to `false` (maybe set to `true` for smaller systems?). Aromaticity of bonds is taken from input data and is calculated calculated for rings (flat and containing only certain types of atoms). Aromaticity is stored separately from bond order (as done in mmcif as well), so you can view the kekulized form, if available, by  setting `aromaticBonds` to false.

Addresses #1 and #178. Ping @stefdoerr.

image

image

### CODE REVIEW

File: src/mol-util/array.ts [hunk 1]
```diff
@@ -105,6 +105,21 @@ export function arraySetRemove<T>(xs: T[], x: T) {
...
+    return false;
+}
+
+export function arrayIntersectionSize<T>(xs: T[], ys: T[]) {
+    let count = 0;
+    for (let i = 0, il = xs.length; i < il; ++i) {
+        if (ys.includes(xs[i])) count += 1;
+    }
+    return count;
+}
```
Comment (dsehnal) [inline src/mol-util/array.ts:121]: I am a little bit worried about the ``O(n^2)`` complexity here. It should be fine for the rings as they are small, but would be careful about leaving the method here as it is. Maybe add a comment at least saying it's for small arrays only?

File: src/mol-repr/structure/visual/util/bond.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ export const BondParams = {
     includeTypes: PD.MultiSelect(ObjectKeys(BondType.Names), PD.objectToOptions(BondType.Names)),
     excludeTypes: PD.MultiSelect([] as BondType.Names[], PD.objectToOptions(BondType.Names)),
     ignoreHydrogens: PD.Boolean(false),
+    aromaticBonds: PD.Boolean(false, { description: 'Display aromatic bonds with dashes' }),
```
Comment (dsehnal) [inline src/mol-repr/structure/visual/util/bond.ts:23]: Why false by default?

**Review by dsehnal — APPROVED** [review]
Looks good, nothing seems to pop out.

File: src/mol-repr/structure/visual/util/bond.ts (hunk 2 above)
Comment (arose) [reply src/mol-repr/structure/visual/util/bond.ts:23]: 1) Don't want to change the current behavior. 2) Not sure what people want to see more, the kekulized form or the aromatics.

File: src/mol-util/array.ts (hunk 1 above)
Comment (arose) [reply src/mol-util/array.ts:121]: Good point, will add a note

**Pull request merged**